### PR TITLE
Ensure generated Kotlin code follow Kotlin naming convention

### PR DESCRIPTION
**Pull request by bmarty**
Constructor parameters cannot start by an upper case letter.
This PR fixes that, hopefully without any side effect on the sent analytics, and on the other generated code.

### CODE REVIEW

File: types/kotlin2/UserProperties.kt
Comment (bmarty) [inline types/kotlin2/UserProperties.kt:46]: This are the problems reported by Detekt, a static analysis tool.

Comment (ouchadam) [reply types/kotlin2/UserProperties.kt:46]: this looks like an issue in the source file https://github.com/matrix-org/matrix-analytics-events/blob/main/schemas/UserProperties.json

I'm not sure if posthog will handle casing as unique properties~

Comment (bmarty) [reply types/kotlin2/UserProperties.kt:46]: That's why I did not change the value we send to Posthog but just the `internal` Kotlin code

Comment (pixlwave) [reply types/kotlin2/UserProperties.kt:46]: I think I’d agree with Adam that this should be fixed in the schema to maintain consistency.

Comment (bmarty) [reply types/kotlin2/UserProperties.kt:46]: OK, did not want to break the existing schema. @novocaine OK for you (as it will impact the Web code)?

Comment (pixlwave) [reply types/kotlin2/UserProperties.kt:46]: Ah, because it’ll change the events sent to PostHog won’t it, yeah I’m guessing we probably don’t want to do that 🫤

Comment (ouchadam) [reply types/kotlin2/UserProperties.kt:46]: in this instance I believe the schema itself is inconsistent and would lean towards it being changed to avoid future confusing but I'm also wondering if the android project shouldn't be enforcing project conventions on the analytics repository? :thinking:

Comment (bmarty) [reply types/kotlin2/UserProperties.kt:46]: I am happy to update the plan if confirmed by the Web maintainer.

Comment (t3chguy) [reply types/kotlin2/UserProperties.kt:46]: Can someone summarise the proposal here?

Comment (bmarty) [reply types/kotlin2/UserProperties.kt:46]: We want to ensure all properties start with a lower case. Applicable to this one https://github.com/matrix-org/matrix-analytics-events/blob/d2ad5869fe479b7b1465b7b2f1f88178e9cc44c0/schemas/UserProperties.json#L23 and following.

Comment (t3chguy) [reply types/kotlin2/UserProperties.kt:46]: It'd probably be better to adjust the generator to skip things for Mobile which are prefixed with `Web` - rather than having to break existing posthog cohorts and lose historical data

Comment (bmarty) [reply types/kotlin2/UserProperties.kt:46]: This PR adjust the Kotlin generator to generate valid Kotlin code without changing the data sent by Posthog. To me it was fine to do that. But I agree that mobile will not send `Web` property. This is something else which could be handled in another PR.

Would be happy to stop spending time on this little change which has *I think* no impact.

Comment (t3chguy) [reply types/kotlin2/UserProperties.kt:46]: The issue is we can't adjust the Typescript generator to tweak this without a lot of work given it is external to this project https://www.npmjs.com/package/json2ts

Comment (bmarty) [reply types/kotlin2/UserProperties.kt:46]: I do not understand why it's preventing this PR to be merged. This PR does not change the plan and do not change the typescript code.

Comment (t3chguy) [reply types/kotlin2/UserProperties.kt:46]: I never said anything here is preventing this PR being merged.

I'm arguing with

> in this instance I believe the schema itself is inconsistent and would lean towards it being changed to avoid future confusion

Comment (bmarty) [reply types/kotlin2/UserProperties.kt:46]: OK, sorry 🤗 . CC @ouchadam ^

Comment (novocaine) [reply types/kotlin2/UserProperties.kt:46]: properties should be camelCase, I think that's fairly clear from examples in the README and in the code.

`Web` has been used as an enum prefix elsewhere which is why it was an uppercase W there, but I don't see why it should be so here, where we are describing a property. It should start with `web`. 

In terms of how disruptive this is to fix, I would be pretty surprised if posthog was case-sensitive..?

**Comment by ouchadam** [comment]
codewise the fix LGTM, there's still a pending question of if this change is needed and sets a precedent for allowing future inconsistent property naming

perhaps we can merge this and create a separate issue to provide a github action to linter to enforce consistency in the json schemas

**Review by ouchadam — APPROVED** [review]

**Pull request merged**
